src/game_actor.cpp

```diff
@@ -114,6 +118,12 @@ bool Game_Actor::IsSkillUsable(int skill_id) const {
 	if (skill_id <= 0 || skill_id > (int)Data::skills.size()) {
 		return false;
 	}
+	for (int inflictedState : this->GetInflictedStates()) {
+		if ((Data::states[inflictedState - 1].restrict_magic && Data::skills[skill_id].magical_rate > Data::states[inflictedState - 1].restrict_magic_level) || ((Data::states[inflictedState - 1].restrict_skill && Data::skills[skill_id-1].hit > Data::states[inflictedState - 1].restrict_skill_level))) {
+			return false;
+		}	
+	}
```

Comment:
This is still of-by-one for skill_id, you use it twice (it's btw hard to find changes after reviewing when you completely rewrite the history^)